cavern/src/main/java/org/opencadc/cavern/files/PutAction.java

```diff
@@ -89,26 +89,27 @@
 import java.security.AccessControlException;
 import java.security.DigestInputStream;
 import java.security.MessageDigest;
-import java.util.Iterator;
 import org.apache.log4j.Logger;
 import org.opencadc.cavern.nodes.NodeUtil;
 
 /**
  *
  * @author majorb
+ * @author jeevesh
  */
-public class PutAction extends FileAction {
+public abstract class PutAction extends FileAction {
     private static final Logger log = Logger.getLogger(PutAction.class);
 
     private static final String INPUT_STREAM = "in";
 
-    public PutAction() {
-        super();
+    public PutAction(Direction pushToVoSpace, boolean isPreauth) {
+        super(pushToVoSpace, isPreauth);
```

Comment:
See my comment in GetAction contructor...

Reply:
ok